cdk/stack.py

```diff
@@ -98,6 +102,76 @@ def __init__(
             description="Name of table used to store ingestions",
         )
 
+        env_secret = self.build_env_secret(config.stage, env)
+        secret_arn: str = env_secret.secret_arn
+
+        self.build_oidc(
+            oidc_provider_arn=config.oidc_provider_arn,
+            oidc_repo_id=config.oidc_repo_id,
+            secret_arn=secret_arn,
+            stage=config.stage,
+        )
+
+    def build_oidc(
+        self, oidc_provider_arn: str, oidc_repo_id: str, secret_arn: str, stage: str
+    ):
+        if oidc_provider_arn:
+            # Create an IAM OIDC provider for the specified provider ARN
+            oidc_provider = iam.OpenIdConnectProvider.from_open_id_connect_provider_arn(
+                self, "OIDCProvider", oidc_provider_arn
+            )
+            # create IAM role for provider access from specified repo
+            # the role should allow a github action in that repo
+            # to deploy resources (TODO) and read a secret
```

Comment:
the `TODO` comment - are we manually adding the permissions to deploy the resources right now?

Reply:
No - I just meant that this repo doesn't have a deployment action set up yet. When we set up this action, we should be able to use [Cloudtrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-user-guide.html) to provide minimum permissions to the role to support deployment.